packages/protocol/src/SavingsContract.ts

```diff
@@ -33,24 +33,32 @@ export function getOrCreateSavingsContract(
   if (version == 2) {
     let tokenEntity = token.getOrCreate(addr)
     savingsContractEntity.token = tokenEntity.id
-  }
-
-  if (massetAddress != null) {
-    savingsContractEntity.masset = massetAddress.toHexString()
-  } else {
-    let underlying = contract.try_underlying()
-    if (underlying.reverted) {
+    let fraction = contract.try_fraction()
+    if (fraction.reverted) {
       throw new Error('Unable to identify underlying for SavingsContract')
     } else {
-      savingsContractEntity.masset = underlying.value.toHexString()
+      savingsContractEntity.fraction = fraction.value
+    }
+
+    if (massetAddress != null) {
```

Comment:
This logic has been moved to an if statement that restricts it to v2 only, but it is required for v1 as well